### PR TITLE
Frontend Updates

### PR DESCRIPTION
**Pull request by vividroyjeong**
- VDYP-460: implement File Upload page
- VDYP-458: Reflecting mockup design into UI development
- VDYP-459: Define the API integration Layer on the Frontend
- Creating custom spin buttons to maintain specific patterns for the number of decimal places per input parameter, and creating handler for the neccsary events
- Applying state updates to buttons and panels based on the user input process
- Implementing screen-speific validations
- Implementing the validations in the libraries referenced by winVDYP into the web UI validations of Stand Density
- Adding the missing Current Diameter field and its associated validations
- Adding field-specific dynamic hints and placeholders
- Adding message dialog and update confirm dialog
- Refactoring for efficiency and consistency in management (e.g. Constantiation Jobs)
- Making design adjustments

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -24,10 +24,13 @@
     "@vitejs/plugin-vue": "^5.0.5",
     "@vue/tsconfig": "^0.5.1",
     "@vuepic/vue-datepicker": "^8.8.0",
-    "axios": "^1.7.2",
+    "axios": "^1.7.7",
```
Comment (dhlevi) [inline frontend/package.json:27]: Ideally, we're using fixed versions. Including the `~` or `^` to allow for auto-updating on build can sometimes lead to unexpected importing

**Review by dhlevi — APPROVED** [review]
The PR is very large. Ideally we're only PR'ing changes of about 500 lines of code. It's very easy to miss things when reviewing this much code.

Please ensure you're cleaning up unused or commented out code. There was a lot of that throughout and if it isn't needed, we can remove it.

Also, there are a lot of cases where we're using null operators but we aren't including any behaviours around null validation. Please make sure you include null checks that correctly handle and throw messages as needed, vs. silently failing.

Overall changes look as expected, but I may want to do some further review to make sure I didn't miss anything

**Comment by vividroyjeong** [comment]
@dhlevi Thank you for your code review! I'll close this PR and create a new PR for the fixes you pointed out and the change to show the results tab on the file upload screen.

**Comment by vividroyjeong** [comment]
@dhlevi I thought it would be easier to reference the code view by reusing this PR, so I didn't close it, but pushed the updated source. Above, you can see a list of additional fixes. Please review the code again.
image

**Review by dhlevi — APPROVED** [review]

**Pull request merged**